### PR TITLE
Add recipe for libusb/hidapi

### PR DESCRIPTION
**Pull request by Jihadist**
Specify library name and version:  hidapi/0.10.1

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Jihadist** [comment]
**Windows**: 0.9.0 cannot build with .vcproj directly so I remove this version. 0.10.0 provides full back compability with 0.9.0 so there is no reason to install older library. 

**Unix**: 
```
[Installing package: hidapi/0.10.0
Requirements
    hidapi/0.10.0 from local cache - Cache
    libusb/1.0.24 from 'conan-center' - Cache
Packages
    hidapi/0.10.0:0114ceff673f5254d496d14b2aa6e2022bc465af - Build
    libusb/1.0.24:a19e97fe6b80c9260c53f0624a793ebe3a192404 - Cache]
```
But configure cannot find libusb
checking for libusb... no
```
  Library libusb-1.0 was not found on this system.
  Please install it and re-run ./configure
```
What i did wrong?

File: recipes/hidapi/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+sources:
+  "0.10.0":
+    url: "https://github.com/libusb/hidapi/archive/hidapi-0.10.0.tar.gz"
+    sha256: "68FEBD416CB6E6E6E205C9DD46A6F86F0D5A9808B7CD8C112906CD229889B8E1"
```
Comment (prince-chrismc) [inline recipes/hidapi/all/conandata.yml:4]: There no need for older versions

```suggestion
```

File: recipes/hidapi/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, MSBuild, tools
+import os, glob
+
+class HidapiConan(ConanFile):
+    name = "hidapi"
+    license = "BSD-Style"
```
Comment (prince-chrismc) [inline recipes/hidapi/all/conanfile.py:6]: Yikes this is complicated https://github.com/libusb/hidapi/blob/master/LICENSE.txt

https://spdx.org/licenses/BSD-3-Clause.html https://spdx.org/licenses/GPL-3.0-or-later

```suggestion
    license = "GPL-3.0-or-later", "BSD-3-Clause", "HIDAPI"
```

File: recipes/hidapi/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        if self.settings.os != "Macos":
+            del self.options.minosx
+        if self.settings.os != "Linux":
+            del self.options.with_libusb
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.settings.os == "Linux" or self.settings.os == "FreeBSD":
+            self.requires("libusb/[~=1.0]")
```
Comment (prince-chrismc) [inline recipes/hidapi/all/conanfile.py:50]: This needs to be an exact version for CCI

Comment (prince-chrismc) [inline recipes/hidapi/all/conanfile.py:50]: These two do no match

maybe allow FreeBSD above and use `self.options.get_safe("...")` ?

File: recipes/hidapi/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    
+    def build(self):
+        if self._is_msvc:
+            self.build_msvc()
+        else:
+            self.build_unix()
+
+    def build_msvc(self):
+        msbuild = MSBuild(self)
+        msbuild.build("%s/windows/hidapi.vcxproj" % self._source_subfolder,
```
Comment (prince-chrismc) [inline recipes/hidapi/all/conanfile.py:80]: Please use `os.path.join`

File: recipes/hidapi/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        if self.settings.os == "Macos":
+            configure = "%s/configure" % self._source_subfolder
+            tools.replace_in_file(configure, r"-install_name \$rpath/",
+                                  "-install_name ")
+        autotools = AutoToolsBuildEnvironment(self)
+        if self.settings.os == "Macos":
+            autotools.flags.append('-mmacosx-version-min=%s' %
+                                   self.options.minosx)
+
+        autotools = self._configure_autotools()
```
Comment (prince-chrismc) [inline recipes/hidapi/all/conanfile.py:96]: This looks like a conflict 😕

File: recipes/hidapi/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+                os.remove(la_file)
+
+    def package(self):
+        self.copy("LICENSE*.txt", src=self._source_subfolder, dst="licenses")
+        self.copy("hidapi/*.h", dst="include", src=self._source_subfolder)
+        self.copy("*hidapi.lib", dst="lib", keep_path=False)
+        self.copy("*.dll", dst="bin", keep_path=False)
+        self.copy("*.so", dst="lib", keep_path=False)
+        self.copy("*.dylib", dst="lib", keep_path=False)
+        self.copy("*.a", dst="lib", keep_path=False)
```
Comment (prince-chrismc) [inline recipes/hidapi/all/conanfile.py:117]:
```suggestion
    def package(self):
        autotools.install()
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "share"))
        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
```

File: recipes/hidapi/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+    def package_info(self):
+        if self.settings.os == "Linux":
```
Comment (prince-chrismc) [inline recipes/hidapi/all/conanfile.py:120]: I see there's a `pkgconfig` file being deleted you need to explicitly set it here

File: recipes/hidapi/config.yml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+versions:
+  "0.10.1":
+    folder: all
+  "0.10.0":
+    folder: all
+  "0.9.0":
+    folder: all
```
Comment (prince-chrismc) [inline recipes/hidapi/config.yml:7]:
```suggestion
  "0.10.1":
    folder: all
```

**Review by prince-chrismc — CHANGES_REQUESTED** [review]

File: recipes/hidapi/all/conanfile.py (hunk 6 above)
Comment (Jihadist) [reply recipes/hidapi/all/conanfile.py:117]: Thanx for this, i didn't think about it

**Comment by prince-chrismc** [comment]
I can not see the failure in the logs, you'll need to test it locally

**Pull request closed without merging**